iam_builder/iam_builder.py

```diff
@@ -20,19 +19,34 @@ def build_iam_policy(config):
     iam = copy.deepcopy(iam_base_template)
 
     list_buckets = []
+
+    # Create lookup dict based on selected or default Athena dump bucket or buckets
+    try:
+        dump_bucket = config["athena"]["dump_bucket"]
+
+        if not isinstance(dump_bucket, list):
+            dump_bucket = [dump_bucket]
+
+        dump_bucket = [bucket.replace("_", "-") for bucket in dump_bucket]
+
+    except KeyError:
+        dump_bucket = ["mojap-athena-query-dump"]
+
+    lookup = iam_lookup(dump_bucket)
+
```

Comment:
I would remove this and move a similar code chunk under the `if "athena"...` statement (line 38). Perhaps this instead:

```python
if "athena" in config:
    dump_bucket = config["athena"].get("dump_bucket", ["mojap-athena-query-dump"])
    if not isinstance(dump_bucket, list):
        dump_bucket = [dump_bucket]
    dump_bucket = [bucket.replace("_", "-") for bucket in dump_bucket]
    lookup = iam_lookup(dump_bucket)
``` 

I think this would make it clearer that this lookup is specific to Athena rather than for everything

Reply:
Done!